pkgs/build-support/php/build-php.nix

```diff
@@ -0,0 +1,106 @@
+{ stdenvNoCC, lib, writeTextDir, composer, php }@inputs:
+
+{ pname
+, version
+, src
+, composerLock ? "__unset"
+, vendorDir ? "vendor"
+, vendorHash
+, php ? inputs.php
+, composer ? inputs.composer
+, ...
+}@args:
+
+let
+  vendor = stdenvNoCC.mkDerivation (finalAttrs: {
+    inherit src version;
+
+    pname = "${pname}-vendor";
+
+    nativeBuildInputs = [
```

Comment:
`unzip` should probably be available to avoid the potential issues mentioned in the Composer FAQ: https://github.com/composer/composer/blob/4e1774518002b0be4365fe62ce220b2011c2b4bf/doc/articles/troubleshooting.md?plain=1#L365-L369

Be aware that `unzip` is locale dependent, see #176253 for more information


Also note that other archive formats require some tools to be present to be unpacked, looking at the code I noted:
* xz: https://github.com/composer/composer/blob/4e1774518002b0be4365fe62ce220b2011c2b4bf/src/Composer/Downloader/XzDownloader.php#L29
* rar: https://github.com/composer/composer/blob/4e1774518002b0be4365fe62ce220b2011c2b4bf/src/Composer/Downloader/RarDownloader.php#L37

Reply:
Added `unzip`, `unrar`, `xz`